### PR TITLE
www/nginx: some bugfixes; add a rule for the upcoming naxsi release; add restart action to acme

### PR DESCRIPTION
**Pull request by fabianfrz**
www/nginx:

* add an if check for rules because it may not exist
* give the www php socket to www
* insert whitelist rule 19 (block if no rules loded) to lower the risk of an incorrectly configured system.
* theme for 404 and 50x

security/acme-client:

* rename SSL certificate to X.509 certificate - this is what is is correctly called - I hope nobody is still using SSL and X.509 is not bound to TLS (can probably be used for IKE as well)
* add the ability to restart nginx

global:

* add PhpStorm files to `.gitignore`

### CODE REVIEW

File: security/acme-client/src/opnsense/mvc/app/models/OPNsense/AcmeClient/AcmeClient.xml [hunk 1]
```diff
@@ -677,6 +677,7 @@
                     <OptionValues>
                         <restart_gui>Restart OPNsense Web UI</restart_gui>
                         <restart_haproxy>Restart HAProxy (OPNsense plugin)</restart_haproxy>
+                        <restart_nginx>Restart nginx (OPNsense plugin)</restart_nginx>
```
Comment (fichtner) [inline security/acme-client/src/opnsense/mvc/app/models/OPNsense/AcmeClient/AcmeClient.xml]: Nginx ;)

File: net/haproxy/Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 PLUGIN_NAME=		haproxy
-PLUGIN_VERSION=		2.7
+PLUGIN_VERSION=		2.8
```
Comment (fichtner) [inline net/haproxy/Makefile]: I'm sorry wtf?

1. don't bump the version, bump the revision
2. this is haproxy, not acme-client

Comment (fichtner) [reply net/haproxy/Makefile]: ps: when there is a revision, you cannot bump the version without clearing the revision... it's pretty simple.

Comment (fabianfrz) [reply net/haproxy/Makefile]: not good, will fix

Comment (fichtner) [reply net/haproxy/Makefile]: I already did

Comment (fabianfrz) [reply net/haproxy/Makefile]: #755 includes the fix - I should not do the merges that late…

**Pull request merged**